crates/sui-indexer-alt-consistent-store/src/metrics.rs

```diff
@@ -0,0 +1,345 @@
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::IS_WRITE_STOPPED,
+                ));
+
+            self.block_cache_usage
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::BLOCK_CACHE_USAGE,
+                ));
+
+            self.block_cache_pinned_usage
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::BLOCK_CACHE_PINNED_USAGE,
+                ));
+
+            self.estimate_table_readers_mem
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::ESTIMATE_TABLE_READERS_MEM,
+                ));
+
+            self.num_immutable_mem_tables
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::NUM_IMMUTABLE_MEM_TABLE,
+                ));
+
+            self.mem_table_flush_pending
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::MEM_TABLE_FLUSH_PENDING,
+                ));
+
+            self.compaction_pending
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::COMPACTION_PENDING,
+                ));
+
+            self.estimate_pending_compaction_bytes
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::ESTIMATE_PENDING_COMPACTION_BYTES,
+                ));
+
+            self.num_running_compactions
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::NUM_RUNNING_COMPACTIONS,
+                ));
+
+            self.num_running_flushes
+                .with_label_values(&[cf_name])
+                .set(cf_property_int_to_metric(
+                    rocks_db,
+                    &cf,
+                    properties::NUM_RUNNING_FLUSHES,
+                ));
+        });
+    }
+}
+
+impl ConsistentStoreMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        ConsistentStoreMetrics {
+            cf_metrics: ColumnFamilyMetrics::new(registry),
+        }
+    }
+}
+
+/// Start periodic task to report metrics defined in `ConsistentStoreMetrics`.
+pub fn start(
```

Comment:
Hmm, why not use the approach we use for the collecting stats from the PG DB connection pool, where we let Prometheus control the polling interval etc?